### DISCUSSION TITLE
Drop Database?

### DISCUSSION THREAD

**Discussion by jmickey** [prisma/prisma#6777] [Q&A] [answered]
Hi,

Just wondering if there is a way, either with `@prisma/client` or with the Prisma CLI to drop a database completely? I tried:

```
await prisma.$queryRaw(`DROP DATABASE ${dbName};`);
```

But I got an error:

```
Raw query failed. Code: `55006`. Message: `db error: ERROR: cannot drop the currently open database`
```

Any advice would be super helpful!

**Comment by pr0ph0z** [comment, marked as answer]
I think this error isn't relying on the Prisma side. And it quite makes sense that you can't drop the database you're currently connected to. Try to connect to a different database and drop the other database (example: connect to database A to drop database B). Don't forget to set proper privileges 😀.

**Reply by pr0ph0z** [reply]
Or if you just want to remove all the tables, just use the `DROP TABLE` command.
POC in MySQL:
```js
const databases = await prisma.$queryRaw(`SELECT concat('DROP TABLE IF EXISTS `', table_name, '`;')
FROM information_schema.tables WHERE table_schema = '${dbName}'`);
await prisma.$executeRaw(`
SET FOREIGN_KEY_CHECKS = 0;
${databases.join(' ')}
SET FOREIGN_KEY_CHECKS = 1;`)
```
reference: https://stackoverflow.com/questions/3476765/mysql-drop-all-tables-ignoring-foreign-keys
Haven't tested that code though, and somehow I'm not suggesting you do that but if you still insist be careful with the code and do the query at your own risk 😀

**Reply by jmickey** [reply]
Thanks very much for the reply. Disconnecting the client and connecting to a different DB did the trick.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
